index.ts

```diff
@@ -0,0 +1,251 @@
+import assert from 'assert';
+import { Mutex } from 'await-semaphore';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
+const EthQuery = require('ethjs-query');
+
+/**
+  *  @property opts.provider - An ethereum provider
+  *  @property opts.blockTracker - An instance of eth-block-tracker
+  *  @property opts.getPendingTransactions - A function that returns an array of txMeta
+  *  whosee status is `submitted`
+  *  @property opts.getConfirmedTransactions - A function that returns an array of txMeta
+  *  whose status is `confirmed`
+*/
+export interface NonceTrackerOptions {
+  provider: Record<string, unknown>;
+  blockTracker: any;
+  getPendingTransactions: (address: string) => Transactions[];
+  getConfirmedTransactions: (address: string) => Transactions[];
+}
+
+/**
+  * @property  highestLocallyConfirmed - A hex string of the highest nonce on a confirmed transaction.
+  * @property  nextNetworkNonce - The next nonce suggested by the eth_getTransactionCount method.
+  * @property  highestSuggested - The maximum between the other two, the number returned.
+  * @property  local - Nonce details derived from pending transactions and highestSuggested
+  * @property  network - Nonce details from the eth_getTransactionCount method
+*/
+export interface NonceDetails {
+  params: {
+    highestLocallyConfirmed: number;
+    nextNetworkNonce: number;
+    highestSuggested: number;
+  };
+  local: HighestContinuousFrom;
+  network: NetWorkNextNonce;
+}
+
+/**
+  * @property  nextNonce - The highest of the nonce values derived based on confirmed and pending transactions and eth_getTransactionCount method
+  * @property  nonceDetails - details of nonce value derivation.
+  * @property  releaseLock
+*/
+export interface NonceLock{
+  nextNonce: number;
+  nonceDetails: NonceDetails;
+  releaseLock: () => void;
+}
+
+/**
+  * @property  name - The name for how the nonce was calculated based on the data used
+  * @property  nonce - The next nonce value suggested by the eth_getTransactionCount method.
+  * @property  blockNumber - The latest block from the network
+  * @property  baseCount - Transaction count from the network suggested by eth_getTransactionCount method
+*/
+export interface NetWorkNextNonce{
```

Comment:
Fix capitalization: `NetWorkNextNonce` -> `NetworkNextNonce` ("network" is one word)